src/ucp/rma/flush.c

```diff
@@ -680,24 +688,56 @@ UCS_PROFILE_FUNC(ucs_status_t, ucp_ep_flush, (ep), ucp_ep_h ep)
 
 UCS_PROFILE_FUNC(ucs_status_t, ucp_worker_fence, (worker), ucp_worker_h worker)
 {
+    ucp_context_t *context          = worker->context;
+    ucp_request_t request;
+    const ucp_request_param_t param = {
+        .op_attr_mask = UCP_OP_ATTR_FIELD_FLAGS | UCP_OP_ATTR_FIELD_REQUEST,
+        .flags        = UCT_FLUSH_FLAG_REMOTE,
+        .request      = UCS_PTR_BYTE_OFFSET(&request, sizeof(request))
+    };
     ucp_rsc_index_t rsc_index;
     ucp_worker_iface_t *wiface;
+    ucs_status_ptr_t status_ptr;
     ucs_status_t status;
 
     UCP_WORKER_THREAD_CS_ENTER_CONDITIONAL(worker);
 
-    UCS_BITMAP_FOR_EACH_BIT(worker->context->tl_bitmap, rsc_index) {
-        wiface = ucp_worker_iface(worker, rsc_index);
-        if (wiface->iface == NULL) {
-            continue;
-        }
+    if (!context->config.worker_fence_eps) {
+        UCS_BITMAP_FOR_EACH_BIT(worker->context->tl_bitmap, rsc_index) {
+            wiface = ucp_worker_iface(worker, rsc_index);
+            if (wiface->iface == NULL) {
+                continue;
+            }
 
-        status = uct_iface_fence(wiface->iface, 0);
-        if (status != UCS_OK) {
+            status = uct_iface_fence(wiface->iface, 0);
+            if (status != UCS_OK) {
+                goto out;
+            }
+        }
+        status = UCS_OK;
+    } else {
+        /* Suppress false positive cpp-checker hits */
+        request.flush_worker.comp_count = 0;
+        request.status                  = UCS_OK;
+        /* force using flush on EPs */
+        ucp_worker_flush_ops_count_add(worker, 1);
+        status_ptr = ucp_worker_flush_nbx_internal(worker, &param);
```

Comment:
will it be difficult to implement ignoring those ep which have just 1 `rma_bw_lane` configured? I think for the fence we do not need to do anything with them